src/python/vehicle.py

```diff
@@ -0,0 +1,183 @@
+    if isinstance(jsonDICT, dict):
+        tag = jsonDICT['tag']
+        contents = jsonDICT['contents']
+
+        if tag == 'Con':
+            def result_func(context):
+                return contents
+
+            return result_func
+
+        elif tag == 'Var':
+            def result_func(context):
+                return context[contents]
+
+            return result_func
+
+        elif tag == 'TensorLit':
+            lossFuncList = []
+            for c in contents:
+                lossFuncList.append(translateJSONtoLOSSfunction(resources, c))
+
+            def result_func(context):
+                tensorOfValues = []
+                for l in lossFuncList:
+                    tensorOfValues.append(l(context))
+                return tf.convert_to_tensor(tensorOfValues)
+
+            return result_func
+
+        elif tag == 'Neg':
+            loss = translateJSONtoLOSSfunction(resources, contents)
+
+            def result_func(context):
+                return -loss(context)
+
+            return result_func
+
+        elif tag == 'Min':
+            loss1 = translateJSONtoLOSSfunction(resources, contents[0])
+            loss2 = translateJSONtoLOSSfunction(resources, contents[1])
+
+            def result_func(context):
+                return min(loss1(context), loss2(context))
+
+            return result_func
+
+        elif tag == 'Max':
+            loss1 = translateJSONtoLOSSfunction(resources, contents[0])
+            loss2 = translateJSONtoLOSSfunction(resources, contents[1])
+
+            def result_func(context):
+                return max(loss1(context), loss2(context))
+
+            return result_func
+
+        elif tag == 'Sub':
+            loss1 = translateJSONtoLOSSfunction(resources, contents[0])
+            loss2 = translateJSONtoLOSSfunction(resources, contents[1])
+
+            def result_func(context):
+                return loss1(context) - loss2(context)
+
+            return result_func
+
+        elif tag == 'Ind':
+            loss1 = translateJSONtoLOSSfunction(resources, contents[0])
+            loss2 = translateJSONtoLOSSfunction(resources, contents[1])
+
+            def result_func(context):
+                if loss1(context) == loss2(context):
+                    return 1
+                else:
+                    return 0
+
+            return result_func
+
+        elif tag == 'At':
+            lossList = contents[0]
+            index = contents[1]
+            lossIndex = translateJSONtoLOSSfunction(resources, index)
+            lossTensor = translateJSONtoLOSSfunction(resources, lossList)
+
+            def result_func(context):
+                return lossTensor(context)[lossIndex(context)]
+
+            return result_func
+            
+        elif tag == 'NetApp':      
+            model = resources[contents[0]]
+            lossInputsList = []
+            for c in contents[1]:
+                lossInputsList.append(translateJSONtoLOSSfunction(resources, c))
```

Comment:
This can be written as list comprehension, e.g.
```python
input_losses = [to_loss_function(resources, c) for c in contents [1]]
```

Indeed, several other cases in this file where a similar refactoring can be done, i.e. anywhere with the pattern:
```python
l = []
for c in ...
  l.append(..)

```